UniversityApp.DataAccess/Repositories/GradesRepository.cs

```diff
@@ -37,15 +37,38 @@ public async Task<List<Grade>> GetAllStudentsGrades()
 
         public async Task<List<Grade>> GetStudentGrades(int studentId)
         {
-            var studentsWithGrades = await _context.Students.Include(g => g.Grades).ToListAsync();
+            var student = await _context.Students.FirstOrDefaultAsync(s => s.Id == studentId);
```

Comment:
Grade entity contains foreign key StudentId, you can get grades from Grade table where studentId has specific value